pkg/policy/selectorcache.go

```diff
@@ -688,33 +678,48 @@ func (sc *SelectorCache) AddIdentitySelector(user CachedSelectionUser, selector
 }
 
 // lock must be held
-func (sc *SelectorCache) removeSelectorLocked(selector CachedSelector, user CachedSelectionUser) {
+func (sc *SelectorCache) removeSelectorLocked(selector CachedSelector, user CachedSelectionUser) (last bool) {
 	key := selector.String()
 	sel, exists := sc.selectors[key]
 	if exists {
-		if sel.removeUser(user, sc.localIdentityNotifier) {
+		if sel.removeUser(user) {
 			delete(sc.selectors, key)
+			return true
 		}
 	}
+	return false
 }
 
 // RemoveSelector removes CachedSelector for the user.
 func (sc *SelectorCache) RemoveSelector(selector CachedSelector, user CachedSelectionUser) {
 	sc.localIdentityNotifier.Lock()
 	sc.mutex.Lock()
-	sc.removeSelectorLocked(selector, user)
+	last := sc.removeSelectorLocked(selector, user)
 	sc.mutex.Unlock()
+	if last {
+		if f, ok := selector.(*fqdnSelector); ok {
+			sc.localIdentityNotifier.UnregisterForIdentityUpdatesLocked(f.selector)
```

Comment:
Maybe this would be better implemented as a fqdn selector specific method, much like the deleted `removeUser()` was, but just for unregistering e.g., `lastUserRemoved(dnsProxy identityNotifier)`?